### PR TITLE
[Enhancement] Add support for discriminators

### PR DESCRIPTION
**Pull request by sserrata**
## Description

Addresses FR #156 

> Note: This implementation of discriminators support requires that discriminator `mapping` be defined. The reason is because we are working with a fully dereferenced OpenAPI object which makes implicit mapping impossible

> Note: This implementation will not support discriminators that are self-referential, i.e. they reference properties within their mapping schema(s). The behavior will be to fall back to `oneOf` if it exists

## Motivation and Context

We should do our best to support all polymorphism types, including discriminators.

## How Has This Been Tested?

See deploy preview Petstore API for examples.

## Screenshots (if appropriate)

Screen Shot 2022-07-26 at 5 53 51 PM

### CODE REVIEW

File: packages/docusaurus-plugin-openapi-docs/src/openapi/utils/loadAndResolveSpec.ts [hunk 1]
```diff
@@ -26,6 +27,18 @@ function serializer(replacer: any, cycleReplacer: any) {
     };
 
   return function (key: any, value: any) {
+    // Resolve discriminator ref pointers
+    if (value.discriminator !== undefined) {
```
Comment (blindaa121) [inline packages/docusaurus-plugin-openapi-docs/src/openapi/utils/loadAndResolveSpec.ts:31]:
```
Loading of api failed for "/Users/blinda/Desktop/pan-docusaurus/docusaurus-openapi/demo/examples/code/CodeSecurity.yaml"
[ERROR] TypeError: Cannot read properties of null (reading 'discriminator')
```
So far tested with the following specs: 
- CSPM 
- CWPP 
- Code Security 
- IoT
- ATP/TP

As for the error message when generating some of the api docs, I added an optional chain in the case that `value` evaluates to null.

`if (value?.discriminator !== undefined) {`
 
 @sserrata I can push the fix up to the same branch if that's fine 👍

Comment (sserrata) [reply packages/docusaurus-plugin-openapi-docs/src/openapi/utils/loadAndResolveSpec.ts:31]: Absolutely, thanks for catching this!

**Comment by sserrata** [comment]
I reordered the description and qualifier and removed the margin-top for consistency.

Screen Shot 2022-07-27 at 8 55 56 AM

I also took one more pass at ensuring there will be nested support for discriminators. Since it's basically handled in `createProperties`, we should support them even when they are nested inside any other schema type, e.g. `allOf`, `anyOf`, etc.

**Comment by sserrata** [comment]
Initially, we were only handling discriminators that mapped directly to a property at the same level as the discriminator. The recent push adds support for discriminators that map to a property inside the discriminator mapping schema itself. 

Example 1:

https://user-images.githubusercontent.com/9343811/181508769-01f20e7a-e5b0-495d-af5b-0d269806856f.mov

Example 2:

https://user-images.githubusercontent.com/9343811/181509011-e0a5361e-77a4-4928-a9ea-6b22c050ed88.mov

**Pull request merged**